subsys/net/lib/dns/resolve.c

```diff
@@ -743,6 +744,13 @@ int dns_validate_msg(struct dns_resolve_context *ctx,
 
 			query_name = dns_msg->msg + dns_msg->query_offset;
 
+			/* Convert the query name to small case so that our
+			 * hash checker can find it.
+			 */
+			for (int i = 0; i < strlen(query_name); i++) {
```

Comment:
Making sure `strlen` gets called only once.

```suggestion
			for (size_t i = 0, n = strlen(query_name); i < n; i++) {
```